src/utils/schemas/SolutionSchema.js

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { z } from 'zod';
+import {
+  CUSTOM_PARAMETER_GROUPS_OPTIONS,
+  CUSTOM_SCENARIO_PARAMETERS_OPTIONS,
+} from './custom/customScenarioParametersOptions';
+
+const getCustomOptionsZodObject = (optionsList) => {
+  let customOptions = {};
+  optionsList.forEach((option) => (customOptions = { ...customOptions, [option]: z.unknown().optional().nullable() }));
+  return z.object({ ...customOptions });
+};
+
+const basicParameterOptions = z.object({
+  enumValues: z
+    .array(
+      z
+        .object({ key: z.string(), value: z.string(), tooltipText: z.object({}).optional().nullable() })
+        .strict()
+        .optional()
+        .nullable()
+    )
+    .optional()
+    .nullable(),
+  tooltipText: z.object({}).optional().nullable(),
+  minLength: z.number().optional().nullable(),
+  maxLength: z.number().optional().nullable(),
+  validation: z.string().optional().nullable(),
+  subType: z.string().optional().nullable(),
+  connectorId: z.string().optional().nullable(),
+  description: z.string().optional().nullable(),
+  defaultFileTypeFilter: z.string().optional().nullable(),
+  canChangeRowsNumber: z.boolean().optional().nullable(),
+  dateFormat: z.string().optional().nullable(),
+  columns: z
+    .array(
+      z
+        .object({
+          field: z.string().optional().nullable(),
+          type: z.array(z.string().optional().nullable()).optional().nullable(),
+          minValue: z.unknown().optional(),
+          maxValue: z.unknown().optional(),
+          acceptsEmptyFields: z.boolean().optional().nullable(),
+          enumValues: z.array(z.string().optional().nullable()).optional().nullable(),
+        })
+        .strict()
+        .optional()
+        .nullable()
+    )
+    .optional()
+    .nullable(),
```

Comment:
Columns groups also need the keys `headerName` and `children` (c.f. [Columns grouping definition](https://github.com/Cosmo-Tech/azure-sample-webapp/blob/main/doc/scenarioParametersConfiguration.md#)).
The value of `children` can be either a column or another column group. Do you think we can easily define this nested structure with zod?

Reply:
I think yes, IIUC object `field` can be declared either directly in `columns` or in `columns.children` ? It will  be a bit verbose but we can declare it in both fields :slightly_smiling_face:

Reply:
Not only `field` but the the other properties too (`type`, `minValue`, ..., `enumValues`).
Maybe we can factorize the schema in a constant `columnOrColumnsGroupSchema` and reuse it?